### PR TITLE
Remove getService API

### LINKED ISSUE #77: Remove getService entrypoint
* Remove `getService()` in favour of the `QIX.connect` (`QIX.create`) API

Depends on #66

### PR DESCRIPTION
**Pull request by peol**
Fixes #77 

The new API, which will most likely change again before release of 2.0 (due to e.g. #64):

Before:

```js
const enigma = require('enigma.js');
enigma.getService('qix', config);
```

After:

```js
const enigma = require('enigma.js');
enigma.connect(config);
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -41,12 +40,12 @@ enigma.js can be used in CommonJS, AMD, and vanilla browser (global) formats. In
...
 ```
 
-Basic usage is to pull in the library, call `getService` and start using the API of the returned service.
+Basic usage is to pull in the library, call `connect` and start using the API of the returned service.
 
 ```javascript
 const enigma = require('enigma.js');
 
-enigma.getService('<Service Name>', { /* Service config */ }).then((serviceInstance) => {
+enigma.connect({ /* Service config */ }).then((serviceInstance) => {
```
Comment (marcusoffesson) [inline docs/README.md]: Should we replace the generic `serviceInstance`with `qix`?

File: docs/README.md [hunk 2]
```diff
@@ -24,12 +24,11 @@ Before you start, you will need to:
...
 enigma.js exposes an API for consuming backend services that are configured with settings.
 
-### `getService( name[, config] )`
+### `connect( config )`
 
 | Property | Type   | Description |
 |----------|--------|-------------|
-| `name`   | String | Name of the service. See the service section for alternatives. |
-| `config` | Object | Service configuration, see docs for respective service. |
+| `config` | Object | Service configuration, see docs for details. |
```
Comment (marcusoffesson) [inline docs/README.md]: Link to configuration.md?

File: docs/README.md [hunk 3]
```diff
@@ -24,12 +24,11 @@ Before you start, you will need to:
 
 enigma.js exposes an API for consuming backend services that are configured with settings.
```
Comment (marcusoffesson) [inline docs/README.md]: ... for consuming the QIX service ... ?

File: docs/README.md [hunk 4]
```diff
@@ -24,12 +24,11 @@ Before you start, you will need to:
...
-### `getService( name[, config] )`
+### `connect( config )`
 
 | Property | Type   | Description |
 |----------|--------|-------------|
-| `name`   | String | Name of the service. See the service section for alternatives. |
-| `config` | Object | Service configuration, see docs for respective service. |
+| `config` | Object | Service configuration, see docs for details. |
 
 Returns an enigma.js service used to communicate with backend services.
```
Comment (marcusoffesson) [inline docs/README.md]: ... used to communicate with a QIX backend service ... ?

File: test/fixtures/local-engine.js [hunk 5]
```diff
@@ -16,7 +15,7 @@ const cfg = {
   },
 };
 
-Registry.getService('qix', cfg).then((qix) => {
+enigma.create(cfg).then((qix) => {
```
Comment (marcusoffesson) [inline test/fixtures/local-engine.js]: connect?

**Comment by peol** [comment]
@marcusoffesson As discussed, I'll create an issue to track documentation consolidation :)

**Pull request merged**